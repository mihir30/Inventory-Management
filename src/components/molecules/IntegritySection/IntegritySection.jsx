import React from 'react'
import "./IntegritySection.scss"

const IntegritySection = ({archivedPartsValue, archivedStoredValue}) => {
  return (
    <div className='integrity-layout'>
    <div className='integrity-template'><span class="integrity-information">Lost units (assigned to fired users): None</span></div>
    <div className='integrity-template'><span class="integrity-information">Broken or deprecated units, assigned to someone: None</span></div>
    <div className='integrity-template'><span class="integrity-information">Units and their parts that belong to different users: None</span></div>
    <div className='integrity-template' data-testid="archived-parts-value">{archivedPartsValue>0?<span class="integrity-information-archived">Archived unit types as parts of other unit types: {archivedPartsValue}</span>
    :<span class="integrity-information">Archived unit types as parts of other unit types: None</span>}</div>
    <div className='integrity-template' data-testid="archived-stored-value">{archivedStoredValue>0?<span class="integrity-information-archived">Archived unit types that are still in store: {archivedStoredValue}</span>
    :<span class="integrity-information">Archived unit types that are still in store: None</span>}</div>
    </div>
  )
}

export default IntegritySection