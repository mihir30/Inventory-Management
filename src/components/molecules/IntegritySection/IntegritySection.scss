.integrity-layout{
    width: 697px;
    height: 508px;
    margin: 20px 0px;
    width: 80vw;
    font-size: 14px;
}
.integrity-template{
    box-sizing: border-box;
    padding: 4px 9pt 20px 15px;
}

.integrity-information{
    color: #248056;
    margin: 20px;
}

.integrity-information-archived{
    color: #FF5C4C;
    margin: 20px;
}

