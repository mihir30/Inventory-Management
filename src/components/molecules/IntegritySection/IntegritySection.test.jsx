import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import IntegritySection from './IntegritySection';
import { render, screen } from '@testing-library/react';

describe("Integrity section Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        ReactDom.render(<BrowserRouter><IntegritySection archivedPartsValue={2} archivedStoredValue={7}></IntegritySection></BrowserRouter>, divEl); 
        
    });

    it('renders the component properly and should display the proper value', () => {
        render(<BrowserRouter><IntegritySection archivedPartsValue={2} archivedStoredValue={7}/></BrowserRouter>);
        expect(screen.getByTestId("archived-parts-value")).toHaveTextContent('Archived unit types as parts of other unit types: 2');
        expect(screen.getByTestId("archived-stored-value")).toHaveTextContent('Archived unit types that are still in store: 7');
    });


    it('should have the right integrity section component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});