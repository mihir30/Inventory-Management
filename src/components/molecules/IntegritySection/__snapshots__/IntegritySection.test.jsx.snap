// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integrity section Component should have the right integrity section component snapshot 1`] = `null`;
